pkg/service/request_enrollment.go

```diff
@@ -70,12 +70,25 @@ func decodeJSONRPCEnrollmentResponse(_ context.Context, res jsonrpc.Response) (i
 	if res.Error != nil {
 		return nil, *res.Error
 	}
-	var result enrollmentResponse
-	err := json.Unmarshal(res.Result, &result)
+
+	// Can't unmarshal errors, as they're interfaces. So we unmarshal to
+	// an intermediary first.
+	var tmpResult = struct {
+		NodeKey     string `json:"node_key"`
+		NodeInvalid bool   `json:"node_invalid"`
+		Err         string `json:"error_code"`
```

Comment:
we could also just update the `enrollmentResponse` type to have a field `ErrorCode`, and a method 
```go
func (e enrollmentResponse) Error() error {
  if e.ErrorCode != nil {
    return errors.New(errorCode)
  }
return nil
}
```
OTOH, that's likely to require more work to ensure we don't break existing things..

Reply:
Yeah. I wasn't sure what all that might break.